### PR TITLE
Supporting Feature Flags

### PR DESCRIPTION
**Pull request by Opoku-Agyemang**
This PR introduces Feature Flags support for Massenergize according to the spec here:
https://paper.dropbox.com/doc/Feature-Flags--BkpUY1IpM~wj4Dr1798_Ku2RAQ-E3KtCfp5ZQN2fgzTjB1DJ
The PR works with [Frontend Admin PR](https://github.com/massenergize/frontend-admin/pull/637)

#### Details 
With this PR, you can now fetch all feature flags turned on for 
1. An admin 
2. A  normal user

#### There are now new routes to
1. Add feature flag 
2. Delete a flag 
3. Just list all the available flags for super-admin display 
4. [Slight changes to Sam's `get_feature_flags`.] Now that route will load all features available to a user, and all the communities they have anything to do with, depending on whether they are on the admin site, or user portal. (Check comments in store/get_feature_flags for breakdown)

#### Testing Steps 
1. The Repo is still in local mode, but check to make sure, then run `... makemigrations` and `migrate` for the new db changes
6. For manual testing, Go to django admin and create a feature flag or step into the [sadmin-feature-flags-implementation ](https://github.com/massenergize/frontend-admin/pull/637) branch on the admin repo to test.

### Up next 
* Test files for new routes I have added ( all features listed above have been manually tested though)

### Note 
* Specific communities, and users that are selected when creating a feature maybe used as "whitelist" or "blacklist" depending on whether the "audience" is set to `"SPECIFIC or ALL_EXCEPT"`\
* Also, the relationship between communities and users on a feature is currently implemented with OR logic. (cant say there is any special reason 💀 🤣) However, if we find the downsides, or see more advantages with AND logic, it can easily be changed.

Feedback wld be great 🤝

### CODE REVIEW

File: src/_main_/settings.py
Comment (Opoku-Agyemang) [inline src/_main_/settings.py:31]: will revert this before merging.  Have it here to prevent mistakenly migrating to dev as this PR requires migrations to test

**Comment by frimpongopoku** [comment]
I have made another branch off this one. I am just adding tiny things. 
I want to make it cater for features that are available to everyone except a select list of communities as well 🤔. 
And also add a route to create the flags from the sadmin frontend.

I will make another PR when thats done, so should we wait and merge that one instead? 🤔

**Review by frimpongopoku — APPROVED** [review]
Works well, just adding adding a few things on. So  lets not merge yet

**Comment by frimpongopoku** [comment]
@BradHN1  @Opoku-Agyemang @Davidjwkinghorn @abdullai-t, made a few tweaks 💀

File: src/api/handlers/feature_flag.py
Comment (Opoku-Agyemang) [inline src/api/handlers/feature_flag.py:98]:
```suggestion
    @super_admins_only
```

File: src/database/models.py
Comment (Opoku-Agyemang) [inline src/database/models.py:3205]: the related names should be consistent.

File: src/_main_/utils/feature_flags/FeatureFlagConstants.py
Comment (Opoku-Agyemang) [inline src/_main_/utils/feature_flags/FeatureFlagConstants.py:1]:
```suggestion
class FeatureFlagConstants:
```

**Comment by frimpongopoku** [comment]
@Opoku-Agyemang @BradHN1 @Davidjwkinghorn I have added the tests, and implemented @Opoku-Agyemang 's feedback. I think we are ready here 🤔

File: src/api/handlers/feature_flag.py
Comment (BradHN1) [inline src/api/handlers/feature_flag.py:21]: change to listForSuperAdmin   (consistent with other routes)

File: src/database/models.py
Comment (BradHN1) [inline src/database/models.py:3195]: please add some comments on the attributes.  What is owner, audience, user_audience, etc.
The migration file in the PR doesn't reflect this version of the model.

File: src/database/models.py
Comment (BradHN1) [inline src/database/models.py:3208]: Null or blank should mean the feature never expires

**Review by BradHN1 — CHANGES_REQUESTED** [review]
Minor changes requested

**Pull request merged**
